qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -1184,25 +1196,36 @@ def update_results(self, record_list: List[ResultRecord]):
         -------
             number of records updated
         """
+        query_ids = [res.id for res in record_list]
+        duplicates = len (query_ids) != len (set(query_ids))
 
-        # try:
-        updated_count = 0
-        for result in record_list:
+        with self.session_scope() as session : 
+            
+            found = session.query(ResultORM).filter(ResultORM.id.in_(query_ids)).all()
+            found_dict = {str(record.id): record for record in found}
 
-            if result.id is None:
-                self.logger.error("Attempted update without ID, skipping")
-                continue
-            with self.session_scope() as session:
+            # try:
+            updated_count = 0
+            for result in record_list:
+
+                if result.id is None:
```

Comment:
This should not be checked above when you find all of the query ids.

Reply:
Not sure on this one, since it was in the original code, and we are still iterating over all the results. Shall I use the dictionary of found items instead?

Reply:
I mean I would check and filter these before you open a session.

Reply:
@codtiger

Reply:
Should I pop the results with ```None``` ids out of the list before opening the session? So I only iterate over valid ones?